floors/enforce.go

```diff
@@ -0,0 +1,246 @@
+package floors
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/currency"
+	"github.com/prebid/prebid-server/exchange/entities"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// Enforce does floors enforcement for bids from all bidders based on floors provided in request, account level floors config
+func Enforce(bidRequestWrapper *openrtb_ext.RequestWrapper, seatBids map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid, account config.Account, conversions currency.Conversions) (map[openrtb_ext.BidderName]*entities.PbsOrtbSeatBid, []error, []*entities.PbsOrtbSeatBid) {
+	rejectionErrs := []error{}
+
+	rejectedBids := []*entities.PbsOrtbSeatBid{}
+
+	requestExt, err := bidRequestWrapper.GetRequestExt()
+	if err != nil {
+		return seatBids, []error{errors.New("Error in getting request extension")}, rejectedBids
+	}
+
+	if !isPriceFloorsEnabled(account, bidRequestWrapper) {
+		return seatBids, []error{errors.New("Floors feature is disabled at account or in the request")}, rejectedBids
+	}
+
+	if isSignalingSkipped(requestExt) || !isValidImpBidFloorPresent(bidRequestWrapper.BidRequest.Imp) {
+		return seatBids, nil, rejectedBids
+	}
+
+	enforceFloors := isSatisfiedByEnforceRate(requestExt, account.PriceFloors.EnforceFloorsRate, rand.Intn)
+	if updateEnforcePBS(enforceFloors, requestExt) {
+		err := bidRequestWrapper.RebuildRequest()
+		if err != nil {
+			return seatBids, []error{err}, rejectedBids
+		}
+	}
+
+	enforceDealFloors := account.PriceFloors.EnforceDealFloors && getEnforceDealsFlag(requestExt)
+	seatBids, rejectionErrs, rejectedBids = enforceFloorToBids(bidRequestWrapper, seatBids, conversions, enforceDealFloors)
+
+	return seatBids, rejectionErrs, rejectedBids
+}
+
+// updateEnforcePBS updates prebid extension in request if enforcePBS needs to be updated
+func updateEnforcePBS(enforceFloors bool, requestExt *openrtb_ext.RequestExt) bool {
+	updateReqExt := false
+
+	prebidExt := requestExt.GetPrebid()
+	if prebidExt == nil {
+		prebidExt = new(openrtb_ext.ExtRequestPrebid)
+	}
+
+	if prebidExt.Floors == nil {
+		prebidExt.Floors = new(openrtb_ext.PriceFloorRules)
+	}
+	floorExt := prebidExt.Floors
+
+	if floorExt.Enforcement == nil {
+		updateReqExt = true
```

Comment:
I think you can delete this `updateReqExt` assignment as well.

Reply:
Yes